app/recipes/routes.py

```diff
@@ -0,0 +1,422 @@
+#!/usr/bin/env python3
+import html
+import math
+import re
+from app import mongo
+from bson.objectid import ObjectId
+from datetime import datetime
+from flask import Blueprint, current_app, render_template, redirect, request, url_for, flash, session, Markup
+from slugify import slugify
+
+
+# --------------------- #
+#    Flask Blueprint    #
+# --------------------- #
+recipes = Blueprint("recipes", __name__)
+
+
+# -------------------- #
+#    DB Collections    #
+# -------------------- #
+allergens_collection = mongo.db.allergens
+desserts_collection = mongo.db.desserts
+measurements_collection = mongo.db.measurements
+recipes_collection = mongo.db.recipes
+users_collection = mongo.db.users
+
+
+# ------------------- #
+#    Global Helper    #
+# ------------------- #
+@recipes.context_processor
+def desserts_total():
+    desserts_count = recipes_collection.count
+    return dict(desserts_count=desserts_count)
+
+
+# ---------------------- #
+#    Helper Functions    #
+# ---------------------- #
+
+# Allergens Dropdown List
+def dropdown_allergens():
+    return sorted([item for allergen in allergens_collection.find() for item in allergen.get("allergen_name")])

```

Comment:
I'm not sure why we are looping the second time over allergen.get... what will this give us 
```
[allergen.get("allergen_name") for allergen in allergens_collection.find()]
```